extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java

```diff
@@ -112,6 +114,19 @@ public RoleListWrapper getSystemRoles() {
     return new RoleListWrapper( systemRoles );
   }
 
+  public RoleListWrapper getAllRoles( boolean excludeAnonymous ) {
+    if ( !excludeAnonymous ) {
+      return getAllRoles();
+    }
+
+    IPentahoSession session = PentahoSessionHolder.getSession();
+    String anonymousRole = PentahoSystem.get( String.class, "singleTenantAnonymousAuthorityName", session );
+    List<String> roles = getAllRoles().getRoles();
+    roles.remove( anonymousRole );
+    return new RoleListWrapper( roles );
```

Comment:
You should instead follow the pattern used for the admin role and receive the anonymous role as a property specified by the resource class. Then, before removing check it's specified (not `null`).

Reply:
Addressed